pangeo_forge_recipes/transforms.py

```diff
@@ -538,14 +521,22 @@ class WriteCombinedReference(beam.PTransform, ZarrWriterMixin):
     output_file_name: str = "reference.json"
 
     def expand(self, references: beam.PCollection) -> beam.PCollection[zarr.storage.FSStore]:
+        # self.target_root in tests are strings so accommodate
+        if isinstance(self.target_root, FSSpecTarget):
```

Comment:
As mentioned below, I might be missing something here. Added to accommodate tests

I don't like accommodating test paths inside the code like this. Is there a reason we use `conftest.py:tmp_target_url` instead of `conftest.py:tmp_target` in `test_end_to_end.py` that actually returns an `FSSPecTarget`?